src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -6926,3 +6926,39 @@ def test_halo_req_no_read_deps(monkeypatch):  # pylint: disable=invalid-name
         _, _ = halo_exchange.required()
     assert ("Internal logic error. There should be at least one read "
             "dependence for a halo exchange" in str(excinfo.value))
+
+
+def test_no_halo_exchange_annexed_dofs():  # pylint: disable=invalid-name
+    '''If a kernel writes to a discontinuous field and also reads from a
+    continuous field then that field reads to its annexed dofs. If the
+    continuous field is written to, into its level1 halo in a previous
```

Comment:
Suggest: "...then that field's annexed dofs are read. If the previous write to this continuous field included these annexed dofs then no halo exchange is required."
If I understand correctly, the level-1 halo includes more than just annexed dofs - hence the fact that looping to 'ncell' is sufficient.

Reply:
I've modified the text. As discussed earlier it is not possible to just compute owned and annexed dofs, you can either compute owned, or owned +l1 halo. One reason for this is to ensure that annexed dofs are always correct by redundantly computing into the l1 halo by default.